api/geth_backend.go

```diff
@@ -1317,44 +1380,100 @@ func (b *GethStatusBackend) GetKeyUIDByMnemonic(mnemonic string) (string, error)
 	return info.KeyUID, nil
 }
 
-func (b *GethStatusBackend) generateOrImportAccount(mnemonic string, customizationColorClock uint64, fetchBackup bool, request *requests.CreateAccount, opts ...params.Option) (*multiaccounts.Account, *settings.Settings, *params.NodeConfig, []*accounts.Account, error) {
+type prepareAccountInput struct {
+	customizationColorClock uint64
+	accountID               string
+	keyUID                  string
+	address                 string
+	mnemonic                string
+	restoringAccount        bool
+	derivedAddresses        map[string]generator.AccountInfo
+	fetchBackup             bool
+	keyStoreDir             string
+	opts                    []params.Option
+}
+
+type accountBundle struct {
+	account        *multiaccounts.Account
+	settings       *settings.Settings
+	nodeConfig     *params.NodeConfig
+	subAccounts    []*accounts.Account
+	chatPrivateKey *ecdsa.PrivateKey
+}
+
+func (b *GethStatusBackend) generateOrImportAccount(mnemonic string, customizationColorClock uint64, fetchBackup bool, request *requests.CreateAccount, opts ...params.Option) (*accountBundle, error) {
 	info, err := b.generateAccountInfo(mnemonic)
 	if err != nil {
-		return nil, nil, nil, nil, err
+		return nil, err
 	}
 
 	keyStoreDir, err := b.initKeyStoreDirWithAccount(request.RootDataDir, info.KeyUID)
 	if err != nil {
-		return nil, nil, nil, nil, err
+		return nil, err
 	}
 
-	account, info, err := b.generateAccount(*info, customizationColorClock, request)
+	derivedAddresses, err := b.getDerivedAddresses(info.ID)
 	if err != nil {
-		return nil, nil, nil, nil, err
+		return nil, err
 	}
 
-	derivedAddresses, err := b.getDerivedAddresses(info.ID)
+	input := &prepareAccountInput{
+		customizationColorClock: customizationColorClock,
+		accountID:               info.ID,
+		keyUID:                  info.KeyUID,
+		address:                 info.Address,
+		mnemonic:                info.Mnemonic,
+		restoringAccount:        mnemonic != "",
+		derivedAddresses:        derivedAddresses,
+		fetchBackup:             fetchBackup,
+		keyStoreDir:             keyStoreDir,
+		opts:                    opts,
+	}
+
+	return b.prepareNodeAccount(request, input)
+}
+
+func (b *GethStatusBackend) prepareNodeAccount(request *requests.CreateAccount, input *prepareAccountInput) (*accountBundle, error) {
+	var err error
+	response := &accountBundle{}
+
+	if request.KeycardInstanceUID != "" {
+		request.Password = input.derivedAddresses[pathEncryption].PublicKey
+	}
+
+	if request.KeycardInstanceUID == "" {
```

Comment:
Actually I left it separately intentionally. 

Because the second statement:
https://github.com/status-im/status-go/blob/bc59afe8ea8049af29abc84dfb6edc3afbd6a786/api/geth_backend.go#L1445-L1448
... is not an "opposite logic" of the first statement:
https://github.com/status-im/status-go/blob/bc59afe8ea8049af29abc84dfb6edc3afbd6a786/api/geth_backend.go#L1441

It's just kinda like this:
1. replace password when we're using keycard
2. store account when we're not using keycard

Does this make sense?

Reply:
I can state this in the code. Or I can replace with an `else` if you think so. Not a big deal for me really 🙂

Reply:
Added a comment to the code to explain the intentions: 381cde1409bf476d38ff4ed97e91824abc86fbd3